core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -557,135 +566,176 @@ public void applyState(final Address sender, int topologyId, boolean pushTransfe
       if (rebalanceTopologyId == NO_STATE_TRANSFER_IN_PROGRESS && !pushTransfer) {
          log.debugf("Discarding state response with topology id %d for cache %s, we don't have a state transfer in progress",
                topologyId, cacheName);
-         return;
+         return CompletableFutures.completedNull();
       }
       if (topologyId < rebalanceTopologyId) {
          log.debugf("Discarding state response with old topology id %d for cache %s, state transfer request topology was %b",
                topologyId, cacheName, waitingForState);
-         return;
+         return CompletableFutures.completedNull();
       }
 
       if (trace) {
          log.tracef("Before applying the received state the data container of cache %s has %d keys", cacheName,
                     dataContainer.sizeIncludingExpired());
       }
-      final CountDownLatch countDownLatch = new CountDownLatch(stateChunks.size());
-      if (pushTransfer) {
-         // push-transfer is specific for scattered cache but this is the easiest way to integrate it
-         for (StateChunk stateChunk : stateChunks) {
-            if (stateChunk.getCacheEntries() != null) {
-               stateTransferExecutor.submit(() -> {
-                  doApplyState(sender, stateChunk.getSegmentId(), stateChunk.getCacheEntries());
-                  countDownLatch.countDown();
-               });
+      IntSet mySegments = IntSets.from(wCh.getSegmentsForOwner(rpcManager.getAddress()));
```

Comment:
Do we need a copy here? It doesn't seem like the `mySegments` is ever modified. Looking at the `getSegmentsForOwner` implementations they are all either a copy or immutable anyways.

Reply:
You're the one who added the copy for ISPN-9293, so I'll trust you and make the change :)

Reply:
Actually no, it's not a copy, just a cast if the CH impl returned an `IntSet`

Reply:
Ah yes, I should have looked at the `IntSets` class again :)